token/program-2022/src/instruction.rs

```diff
@@ -409,6 +427,125 @@ pub enum TokenInstruction {
+    InitializePaymentMint {
+        /// Number of base 10 digits to the right of the decimal place.
+        decimals: u8,
+        /// The authority/multisignature to mint tokens.
+        mint_authority: Pubkey,
+        /// The freeze authority/multisignature of the mint.
+        freeze_authority: COption<Pubkey>,
+        /// Pubkey that must own any provided fee account
+        fee_account_owner: Pubkey,
+        /// Amount of transfer collected as fees, expressed as basis points of the
+        /// transfer amount
+        transfer_fee_basis_points: u16,
+        /// Maximum fee assessed on transfers
+        maximum_fee: u64,
+    },
+    /// Initializes a new `PaymentAccount` to hold tokens.  The mint provided must
+    /// be a `PaymentMint`, and the account must have enough data allocated for
+    /// a `PaymentAccount` instance.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[writable]`  The account to initialize.
+    ///   1. `[]` The mint this account will be associated with. Must be a `PaymentMint`.
+    InitializePaymentAccount {
+        /// The new account's owner/multisignature.
+        owner: Pubkey,
+    },
+    /// Transfer, providing expected mint information and fees
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   * Single owner/delegate
+    ///   0. `[writable]` The source account. Must be a PaymentAccount.
+    ///   1. `[]` The token mint. Must be a PaymentMint.
+    ///   2. `[writable]` The destination account. Must be a PaymentAccount.
+    ///   3. `[signer]` The source account's owner/delegate.
+    ///
+    ///   * Multisignature owner/delegate
+    ///   0. `[writable]` The source account.
+    ///   1. `[]` The token mint.
+    ///   2. `[writable]` The destination account.
+    ///   3. `[]` The source account's multisignature owner/delegate.
+    ///   4. ..4+M `[signer]` M signer accounts.
+    TransferCheckedWithFee {
+        /// The amount of tokens to transfer.
+        amount: u64,
+        /// Expected number of base 10 digits to the right of the decimal place.
+        decimals: u8,
+        /// Amount of transfer collected as fees, expressed as basis points of the
+        /// transfer amount
+        transfer_fee_basis_points: u16,
+        /// Maximum fee assessed on transfers
+        maximum_fee: u64,
+    },
+    /// Transfer all withheld tokens to a fee account. Signed by the mint's
+    /// fee withdraw authority.
+    ///
+    /// Succeeds for frozen accounts.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   * Single owner/delegate
+    ///   0. `[writable]` The token mint. Must be a PaymentMint.
+    ///   1. `[writable]` The fee receiver account. Must be a PaymentAccount
+    ///      associated with the provided PaymentMint.
+    ///   2. `[signer]` The mint's `fee_withdraw_authority`
+    ///   3. ..3+N `[writable]` The source accounts to harvest from. Must be PaymentAccounts.
```

Comment:
`3..` is optional right?  That is, if no `PaymentAccounts` are provided then any withheld tokens in the token mint are still transferred to the fee receiver account

Reply:
Yep, I'll make that clearer